pilot/pkg/serviceregistry/mcp/controller.go

```diff
@@ -108,18 +109,32 @@ func (c *controller) List(typ resource.GroupVersionKind, namespace string) (out
 		// we replace the entire sub-map
 		for _, byNamespace := range byType {
 			for _, config := range byNamespace {
-				out = append(out, *config)
+				if c.objectInRevision(config) {
```

Comment:
Why donot check this before caching these configs?

Reply:
I have change to do this in Apply